Sources/HTMLString/HTMLString.swift

```diff
@@ -37,23 +35,27 @@ extension String {
     /// | `a` | `a` | Not escaped (alphanumerical) |
     ///
 
-    public mutating func addUnicodeEntities() {
+    public func addUnicodeEntities() -> String {
         var position: String.Index? = startIndex
         let requiredEscapes: Set<Character> = ["!", "\"", "$", "%", "&", "'", "+", ",", "<", "=", ">", "@", "[", "]", "`", "{", "}"]
+        var result = ""
```

Comment:
Let's reserve the capacity here so that we can avoid needing to expand it when iterating.

```suggestion
        var result = String()
        result.reserveCapacity(count)
```

Reply:
It is good idea, but the problem is that time complexity of String's count is O(n). Although it isn't intuitive, It's true because the size of UTF-8 characters is varies. Here's the test result of reserving capacity.(testLargeUnescapingPerformance)

<img width="235" alt="스크린샷 2020-09-26 오전 9 06 29" src="https://user-images.githubusercontent.com/51935215/94325074-ae90d400-ffd7-11ea-9aa0-a478e1489e09.png">

and here is an official documentation. String is not RandomAccessCollection, but BidirectionalCollection.

https://developer.apple.com/documentation/swift/collection/3017670-count

Reply:
Fair enough, I wasn't sure if not reserving the capacity would result in a worse performance. Let's not do it then.